docs/building-a-form/about-the-schema-and-uischema-objects.md

```diff
@@ -0,0 +1,222 @@
+  },
+  'ui:options': {
+
+    // An map of enum values to labels that are shown by the select and radio widgets.
+    labels: {
+      chapter30: 'A readable description (Chapter 30)'
+    },
+
+    // A map of values to a component, text, or JSX (https://reactjs.org/docs/introducing-jsx.html). If your field is a radio widget,
+    // the content here is shown underneath the radio button for that value when it's selected.
+    nestedContent: {
+      'value': <p>Some text</p>
+    },
+
+    // A string of class names that are added to the widget for the current field. `widgetClassNames` is similar to the default
+    // `classNames` property, but it puts the class names on the input/select/etc element itself, rather than a surrounding `div`.
+    widgetClassNames: '',
+
+    // For array fields, this component is shown when the item in the array is rendered as read-only on a page that is not a review page.
+    viewField: RowViewComponent,
+
+    // To show a field only when another field is true, set this option to the property name. It wraps the fields with an
+    // ExpandingGroup component using the `expandUnder` field as the first question.
+    expandUnder: '',
+
+    // To match to a specific value, use the `expandUnderCondition` option to specify the value that the `expandUnder` field's data should equal.
+    expandUnderCondition: 'someValue',
+    // `expandUnderCondition` can also be a function that receives the data from the `expandUnder` field as an argument.
+    expandUnderCondition: (field) => field === 'someValue' || field === 'someOtherValue',
+
+    // When using the expandUnder option, you can set `expandUnderClassNames` on the field specified by `expandUnder` and it will add
+    // classes to the `div` that wraps all of the fields when they're expanded. See cookbook for an example use case.
+    expandUnderClassNames: '',
+
+    // Hides the specified field on the review page.
+    hideOnReview: true || false,
+
+    // Hides the specified field on the review page when the field value is `false`.
+    hideOnReviewIfFalse: true || false
+
+    // A function that conditionally hides fields in the form. `hideIf` provides the `index` argument when you use `ui:required` on data inside an array.
+    hideIf: function (formData, index) {
+      return true || false;
+    }
+
+    // A function that conditionally replaces the current field's schema. `updateSchema` provides the `index` argument when you use `ui:required` on data inside an array.
+    updateSchema: function (formData, schema, uiSchema, index, pathToCurrentData) {
+      // This function returns an object with the properties you want to update. Instead of replacing the existing schema, it updates the individual properties.
+      return {
+        type: 'string'
+      };
+    },
+
+    // Use this when you have an array field that should not be pulled out of the page its in and shown separately on the review page.
```

Comment:
I don't understand this comment but it seems like a concept that would be hard to explain so I wonder if we should actually try to explain it or just reference some other documentation on it.

Reply:
@annekainicUSDS - any insight here?

Reply:
We have default functionality to show array fields as a separate "page" on the review screen. So, for example, on the health care application we ask for your dependent information all on one page:
<img width="818" alt="screen shot 2018-07-19 at 9 20 47 am" src="https://user-images.githubusercontent.com/25183456/42944953-0dfb43b8-8b35-11e8-9ab1-265f611e5df2.png">

But on the review screen the default functionality is to great a separate section, or page, for that information (see the two headers?):
<img width="716" alt="screen shot 2018-07-19 at 9 20 01 am" src="https://user-images.githubusercontent.com/25183456/42944968-1b360194-8b35-11e8-816e-60c29437b7ee.png">

This property keeps the content of an array field within the same section on the review page, at least that's how I'm reading it.

Reply:
A picture is worth a thousand words!